terraform/deployments/github/imports3.tf

```diff
@@ -0,0 +1,532 @@
+import {
+  to = github_team_repository.govuk_repos["govuk-fastly"]
+  id = "3279243:govuk-fastly"
+}
+import {
+  to = github_team_repository.govuk_repos["maslow"]
+  id = "3279243:maslow"
+}
+import {
+  to = github_team_repository.govuk_repos["govuk_test"]
+  id = "3279243:govuk_test"
+}
+import {
+  to = github_team_repository.govuk_repos["support"]
+  id = "3279243:support"
+}
+import {
+  to = github_team_repository.govuk_repos["smokey"]
+  id = "3279243:smokey"
+}
+import {
+  to = github_team_repository.govuk_repos["govuk-pact-broker"]
+  id = "3279243:govuk-pact-broker"
+}
+import {
+  to = github_team_repository.govuk_repos["govuk-display-screen"]
+  id = "3279243:govuk-display-screen"
+}
+import {
+  to = github_team_repository.govuk_repos["rails_translation_manager"]
+  id = "3279243:rails_translation_manager"
+}
+import {
+  to = github_team_repository.govuk_repos["ckanext-datagovuk"]
+  id = "3279243:ckanext-datagovuk"
+}
+import {
+  to = github_team_repository.govuk_repos["ckan-mock-harvest-sources"]
+  id = "3279243:ckan-mock-harvest-sources"
+}
+import {
+  to = github_team_repository.govuk_repos["content-data-api"]
+  id = "3279243:content-data-api"
+}
+import {
+  to = github_team_repository.govuk_repos["packager"]
+  id = "3279243:packager"
+}
+import {
+  to = github_team_repository.govuk_repos["govuk_crawler_worker"]
+  id = "3279243:govuk_crawler_worker"
+}
+import {
+  to = github_team_repository.govuk_repos["whitehall"]
+  id = "3279243:whitehall"
+}
+import {
+  to = github_team_repository.govuk_repos["feedback"]
+  id = "3279243:feedback"
+}
+import {
+  to = github_team_repository.govuk_repos["email-alert-frontend"]
+  id = "3279243:email-alert-frontend"
+}
+import {
+  to = github_team_repository.govuk_repos["local-links-manager"]
+  id = "3279243:local-links-manager"
+}
+import {
+  to = github_team_repository.govuk_repos["service-manual-publisher"]
+  id = "3279243:service-manual-publisher"
+}
+import {
+  to = github_team_repository.govuk_repos["govuk-helm-charts"]
+  id = "3279243:govuk-helm-charts"
+}
+import {
+  to = github_team_repository.govuk_repos["datagovuk-visual-regression-tests"]
+  id = "3279243:datagovuk-visual-regression-tests"
+}
+import {
+  to = github_team_repository.govuk_repos["tagging-suggester"]
+  id = "3279243:tagging-suggester"
+}
+import {
+  to = github_team_repository.govuk_repos["gds_zendesk"]
+  id = "3279243:gds_zendesk"
+}
+import {
+  to = github_team_repository.govuk_repos["govuk-connect"]
+  id = "3279243:govuk-connect"
+}
+import {
+  to = github_team_repository.govuk_repos["govspeak-preview"]
+  id = "3279243:govspeak-preview"
+}
+import {
+  to = github_team_repository.govuk_repos["govuk-technical-2ndline-dashboard"]
+  id = "3279243:govuk-technical-2ndline-dashboard"
+}
+import {
+  to = github_team_repository.govuk_repos["publishing-api"]
+  id = "3279243:publishing-api"
+}
+import {
+  to = github_team_repository.govuk_repos["support-api"]
+  id = "3279243:support-api"
+}
```

Comment:
Sure, I just didn't think it mattered since they will be removed as soon as this gets applied, but will do that.

Reply:
Ah apologies, I didn't spot that these files would be deleted. Can you explain the reasoning for that?
I'm not clear what that would achieve other than one-off applying the correct level of team access to each repo 🤔 
A core principle of Infrastructure as Code is to have something you can reapply again and again, so this doesn't feel quite right.

I'd also want to think about what we'd ask developers to do when creating a _new_ repo - it should be as simple as adding a repo name to an array and clicking an "Apply Terraform" button.

Reply:
Have we definitely confirmed we need Terraform Import rather than just managing existing repo access using the GitHub provider? https://registry.terraform.io/providers/integrations/github/latest/docs/resources/repository_collaborator.html

Reply:
So the reason we need the imports here is that we're adding resources that already exist and were not created using Terraform, so if we don't import them Terraform will try to create them again. Once that gets applied we can remove the imports because Terraform will already be tracking those from then on. I'm not a TF expert though and that's what I learned from talking to people smarter than me.

Reply:
Thanks, that makes sense, but I think it's lacking a 'next step' after the import. By deleting the import.tf files, won't Terraform lose the references to those repos?

govuk-infrastructure already has a mechanism for fetching all repos with the `govuk` tag and applying permissions to push OCI images: https://github.com/alphagov/govuk-infrastructure/blob/main/terraform/deployments/github/main.tf
When I wrote this card, I imagined we'd probably just piggy-back onto that mechanism.

That way, it should work dynamically with any repo that is tagged `govuk`, and we also wouldn't need to get devs to do anything extra other than tag their repo.

Reply:
So, in this case, we're not importing the repos themselves, we're importing the [github teams](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team) and the [github team repos](https://registry.terraform.io/providers/integrations/github/latest/docs/resources/team_repository.html), which is the relationship between teams and repos.

Reply:
I think the best way to review this is to check the code against the [latest run](https://app.terraform.io/app/govuk/workspaces/GitHub/runs/run-ow91YLTVBYVrdyQD) on the Github workspace

Reply:
> It's the same thing we did [here](https://github.com/alphagov/govuk-user-reviewer/pull/972/files) and removed [here](https://github.com/alphagov/govuk-user-reviewer/pull/973/files) or in this repo we added imports in [this pr](https://github.com/alphagov/govuk-infrastructure/pull/925/files) and removed them [here](https://github.com/alphagov/govuk-infrastructure/pull/928/files)

Ah yes, thanks for digging those out, that makes sense. With the latter example above, there's still a reference to each Sentry project [here](https://github.com/alphagov/govuk-infrastructure/blob/main/terraform/deployments/sentry/locals.tf), so it's clear those are being managed in this repo. But it sounds like with the current PR, once the import files are removed, there will be no references to the repo names anymore - am I missing something?

Also can you explain what the new process will be for developers creating new repos - will they need to have an import and revert step for their repo like you're doing en-masse here?

Reply:
(PS I've experimented with not importing the repos, in #943 - agree your Terraform plan output looks better 👍 Though [I wonder what would happen](https://github.com/alphagov/govuk-infrastructure/pull/943#issuecomment-1715997429) after the initial `apply` in my branch... )

Reply:
The import is only needed if the resource already exists, for new repos they will just need to be added in terraform first instead of created manually.